### PR TITLE
Feature/regex principal builder

### PR DESCRIPTION
**Pull request by SgtPepperLHCB**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Docker images for deploying and running Kafka services based on the [Confluent P
 The images are based on [confluentinc/cp-docker-images](https://github.com/confluentinc/cp-docker-images) and have been modified to:
 - [x] Execute as non-privileged users
 - [x] Include a [Prometheus JMX exporter](https://github.com/prometheus/jmx_exporter)
-- [ ] Include a [customized PrincipalBuilder](https://docs.confluent.io/current/kafka/authorization.html#user-names)
+- [x] Include a [customized PrincipalBuilder](https://docs.confluent.io/current/kafka/authorization.html#user-names) (see [RegexPrincipalBuilder](https://github.com/Nordstrom/kafka-regex-principal-builder))
```
Comment (dylanmei) [inline README.md]: Replace the Confluent link between parens `()` in its entirety.

File: broker/Dockerfile [hunk 2]
```diff
@@ -15,4 +15,8 @@ RUN mkdir -p /etc/jmx_exporter /opt/lib \
  && echo "---" > $PROMETHEUS_JMX_EXPORTER_CONF \
  && sed -i "s/^EXTRA_ARGS=.*/EXTRA_ARGS=\${EXTRA_ARGS-'-name kafkaServer -loggc -javaagent:\/opt\/lib\/jmx_prometheus_javaagent.jar='\$PROMETHEUS_JMX_EXPORTER_PORT:\$PROMETHEUS_JMX_EXPORTER_CONF}/" /usr/bin/kafka-server-start
 
+# Add customer principal builder and set permissions on jar file for read by group/others.
```
Comment (dylanmei) [inline broker/Dockerfile]: "customer principal builder" can be "customized PrincipalBuilder"

File: README.md (hunk 1 above)
Comment (SgtPepperLHCB) [reply README.md]: Done

File: broker/Dockerfile [hunk 3]
```diff
@@ -15,4 +15,8 @@ RUN mkdir -p /etc/jmx_exporter /opt/lib \
...
+ADD https://github.com/Nordstrom/kafka-regex-principal-builder/releases/download/v0.0.3/kafka-regex-principal-builder-0.0.3.jar /usr/share/java/kafka/
+RUN chmod 644 /usr/share/java/kafka/kafka-regex-principal-builder-0.0.3.jar
```
Comment (dylanmei) [inline broker/Dockerfile]: Consider also keeping this as a single image layer with:

```
RUN curl -L https://github.com/Nordstrom/kafka-regex-principal-builder/releases/download/v0.0.3/kafka-regex-principal-builder-0.0.3.jar -o /usr/share/java/kafka/ && chmod 644...
```

File: broker/Dockerfile (hunk 2 above)
Comment (SgtPepperLHCB) [reply broker/Dockerfile]: Grrrr. `customer` must be a muscle-memory thing from the days when I worked heavily on Enterprise Domain Models.

File: connect/Dockerfile [hunk 4]
```diff
@@ -14,4 +14,7 @@ RUN mkdir -p /etc/jmx_exporter /opt/lib \
  && sed -i "s/^EXTRA_ARGS=.*/EXTRA_ARGS=\${EXTRA_ARGS-'-name connectStandalone -javaagent:\/opt\/lib\/jmx_prometheus_javaagent.jar='\$PROMETHEUS_JMX_EXPORTER_PORT:\$PROMETHEUS_JMX_EXPORTER_CONF}/" /usr/bin/connect-standalone \
  && sed -i "s/^EXTRA_ARGS=.*/EXTRA_ARGS=\${EXTRA_ARGS-'-name connectDistributed -javaagent:\/opt\/lib\/jmx_prometheus_javaagent.jar='\$PROMETHEUS_JMX_EXPORTER_PORT:\$PROMETHEUS_JMX_EXPORTER_CONF}/" /usr/bin/connect-distributed
 
+#ADD connector jar files on CLASSPATH; /usr/share/java/kafka-connect-NNN is recommended loc.
```
Comment (dylanmei) [inline connect/Dockerfile]: Nice idea to think of future, but this can be removed from code and added to the details of a separate issue/story.

**Comment by dylanmei** [comment]
LGTM!

**Pull request merged**